cumulus_library_opioid/vocab/rxnorm_vsac_builder.py

```diff
@@ -19,65 +19,90 @@ def prepare_queries(
-            b_join_col = None,
-            a_schema = None,
-            view_name = None,
-            **kwargs,
+            steward: str,
+            a_table: str,
+            *,
+            columns : list,
+            b_table:str | None= None,
+            a_join_col: str | None= None,
+            b_join_col: str | None= None,
+            a_schema: str | None= None,
+            view_name:str | None= None,
+            join_clauses: list | None = None,
+            column_aliases: dict | None = None
+            
         ):
             a_schema = a_schema or 'rxnorm.'
-            return base_templates.get_base_template(
-                'create_view_from_tables',
-                self.base_path / "template_sql",
+            return base_templates.get_create_view_from_tables(
                 view_name=view_name or (
                     f'{manifest.get_study_prefix()}__{steward}_{a_table}'
                 ),
-                a_table = f'{a_schema}{a_table}',
-                a_schema = a_schema,
-                b_table= b_table or f'opioid__{steward}_vsac',
-                a_join_col= a_join_col or 'rxcui',
-                b_join_col = b_join_col or 'code',
-                **kwargs
+                tables = [
+                    f'{a_schema}{a_table}',
+                    b_table or f'opioid__{steward}_vsac',
+                ],
+                table_aliases = ['a','b'],
+                join_clauses=join_clauses or [f"{a_join_col or 'a.rxcui'} = {b_join_col or 'b.code'}"],
+                columns=columns,
+                column_aliases = column_aliases,
+
             )
         stewards = vsac.get_vsac_stewards(config)
         for steward in stewards:
             self.queries.append(
                 get_create_view_filter_by(
                     steward,
                     "rxnconso",
-                    a_cols=['rxcui','str','tty','sab','code'],
+                    columns=['a.rxcui','a.str','a.tty','a.sab','a.code'],
+                )
+            )
+            self.queries.append(
+                get_create_view_filter_by(
+                    steward,
+                    f'{steward}_rxnconso',
+                    a_schema=f'{manifest.get_study_prefix()}__',
+                    b_table = 'opioid__keywords',
+                    columns=['a.rxcui','a.str','a.tty','a.sab','a.code', 'b.str'],
+                    view_name=f'{manifest.get_study_prefix()}__{steward}_rxnconso_keywords',
+                    join_clauses = ["lower(a.str) LIKE concat('%',b.STR, '%')"],
+                    column_aliases = {"b.str": "keyword"}
                 )
             )
+            # TODO: Fix this template to mimic the static rxnconso file
+            # self.queries.append(
+            #     base_templates.get_base_template(
+            #         'create_annotated_rxnconso',
+            #         self.base_path / "template_sql",
+            #         steward=steward
+            #     )
```

Comment:
This has been fussy enough that I wanted to get this in and then come back and do the next step as a seperate effort - you could talk me into trying to do it now, but that was my thought.